interop-testing/src/main/java/io/grpc/testing/integration/TestServiceImpl.java

```diff
@@ -202,9 +202,29 @@ public void onError(Throwable cause) {
   public StreamObserver<Messages.StreamingOutputCallRequest> fullDuplexCall(
       final StreamObserver<Messages.StreamingOutputCallResponse> responseObserver) {
     final ResponseDispatcher dispatcher = new ResponseDispatcher(responseObserver);
-    return new StreamObserver<StreamingOutputCallRequest>() {
+    ServerCallStreamObserver<Messages.StreamingOutputCallResponse> autoUnlockResponseObserver =
+        (ServerCallStreamObserver<Messages.StreamingOutputCallResponse>) responseObserver;
+
+    class MayBlockStreamObserver implements StreamObserver<StreamingOutputCallRequest> {
+      boolean oobTestLocked;
+
       @Override
       public void onNext(StreamingOutputCallRequest request) {
+        // to facilitate multiple clients running orca_oob test in parallel, the server allows
+        // only one orca_oob test client to run at a time to avoid conflicts.
+        if (request.hasOrcaOobReport()) {
+          if (!oobTestLocked) {
+            try {
+              lock.acquire();
+            } catch (InterruptedException ex) {
+              autoUnlockResponseObserver.onError(new StatusRuntimeException(
+                  Status.ABORTED.withDescription("server service interrupted").withCause(ex)));
+            }
+            oobTestLocked = true;
+          }
+          echoMetricsFromPayload(request.getOrcaOobReport());
+          responseObserver.onNext(StreamingOutputCallResponse.getDefaultInstance());
```

Comment:
Delete this. Let the `dispatcher.enqueue(toChunkQueue(request));` handle the response.

Reply:
> we need to set ResponseParameters then, looks hacky

In what way? To me this looks hacky. With this implementation it seems it shouldn't be part of the `fullDuplexCall()` method because it is completely unrelated to all the other fields. Instead, we should create a new method for it.

If we let the rest of the code run like it normally does, then we can mix-and-match behavior as we want. Chunkqueue itself we don't care about too much, but that is used for all responses normally; if you don't add a delay for the response it essentially becomes a noop and responds to the RPC immediately.